website/content/posts/introducing-action-validate-image.md

```diff
@@ -0,0 +1,119 @@
+
+ ### Sign image with GitHub OIDC Token
+In this step, the container image is signed using Cosign and a GitHub OIDC token. This adds an additional layer of security and trust to the image, making it easier to verify its integrity and origin.
+
+```yaml
+- name: Sign image with GitHub OIDC Token
+  run: cosign sign --yes ${IMAGE_REGISTRY}/${IMAGE_REPO}@${DIGEST}
+  env:
+    DIGEST: ${{ steps.push-image.outputs.digest }}
+```
+
+### Generate and Store SBOM
+
+We use Syft to generate a SBOM. It's then attested using Cosign, which serves as both documentation and an extra security measure.
+
+```yaml
+- name: Generate and store SBOM
+  run: |
+      syft "${IMAGE_REGISTRY}/${IMAGE_REPO}@${DIGEST}" -o spdx-json=sbom-spdx.json
+      cosign attest --yes --predicate sbom-spdx.json --type spdx "${IMAGE_REGISTRY}/${IMAGE_REPO}@${DIGEST}"
+  env:
+    DIGEST: ${{ steps.push-image.outputs.digest }}
+```
+
+### SLSA Provenance Generation
+We employ SLSA tooling to generate provenance. This helps track the build process and adds traceability to the container image.
+```yaml
+- name: slsa-github-generator
+  uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.9.0 
+  with:
+    image: ${{ needs.build.outputs.image }}
+    digest: ${{ needs.build.outputs.digest }}
+    registry-username: ${{ github.actor }}
+  secrets:
+    registry-password: ${{ secrets.GITHUB_TOKEN }}
+```
+
+### EC Action Validate
+EC Action Validate works by assessing your container images against a set of validation checks. You can customize these checks through a policy to align with the specific security and compliance guidelines of your organization or industry. Whether the image passes or fails, you'll receive a GitHub summary output, and additional logs will be available in YAML format within the action.
+```yaml
+- name: Validate image (keyless)
+  uses: enterprise-contract/action-validate-image@v1.0.31 
+  with:
+    image: ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
+    identity: https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|${{ github.repository_owner }}\/${{ github.event.repository.name }})\/
+    issuer: https://token.actions.githubusercontent.com
+    policy: github.com/enterprise-contract/config//github-default
+```
+- **`image`**: Specifies the container image to be validated. It uses the image and digest output from a previous build step in the workflow.
+
+- **`identity`**: Defines the identity or entities that are allowed to perform the validation. In this case, it's either from the SLSA GitHub Generator or the repository owner.
```

Comment:
"... that are expected to have produced the image signature and attestations." ?